### PR TITLE
Update to Leshan 1.0.0-M1

### PR DESCRIPTION
**Pull request by DanielMaier-BSI**
Hi,

as promised in our last PR, here the update to leshan 1.0.0-M1 for the LWM2M branch.

Signed-off-by: Daniel Maier <Daniel.Maier@bosch-si.com>

### CODE REVIEW

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java [hunk 1]
```diff
@@ -61,51 +56,50 @@
...
 
     @Autowired
-    public void setClientRegistry(final ClientRegistry clientRegistry) {
-        this.clientRegistry = Objects.requireNonNull(clientRegistry);
-        LOG.info("using ClientRegistry: {}", clientRegistry.getClass().getName());
-    }
-
-    @Autowired
-    public void setModelProvider(final LwM2mModelProvider modelProvider) {
+    public void setModelProvider(LwM2mModelProvider modelProvider) {
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java]: make param final

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java [hunk 2]
```diff
@@ -61,51 +56,50 @@
...
         this.modelProvider = Objects.requireNonNull(modelProvider);
         LOG.info("using LwM2mModelProvider: {}", modelProvider.getClass().getName());
     }
 
     @Autowired
-    public void setLeshanConfigProperties(final LeshanConfigProperties leshanConfig) {
+    public void setLeshanConfigProperties(LeshanConfigProperties leshanConfig) {
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java]: param final

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java [hunk 3]
```diff
@@ -61,51 +56,50 @@
...
         this.leshanConfig = Objects.requireNonNull(leshanConfig);
     }
 
     @Autowired
-    public void setSecurityRegistry(final SecurityRegistry securityRegistry) {
-        this.securityRegistry = Objects.requireNonNull(securityRegistry);
-        LOG.info("using SecurityRegistry: {}", securityRegistry.getClass().getName());
+    public void setSecurityStore(EditableSecurityStore securityStore) {
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java]: param final

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java [hunk 4]
```diff
@@ -61,51 +56,50 @@
...
+        this.securityStore = Objects.requireNonNull(securityStore);
+        LOG.info("using SecurityStore: {}", securityStore.getClass().getName());
     }
 
-    @Autowired(required = false)
-    public void setClientRegistryListeners(final Set<ClientRegistryListener> listeners) {
-        Objects.requireNonNull(listeners);
-        this.clientRegistryListeners.addAll(listeners);
+    @Autowired
+    public void setServerKeyProvider(ServerKeyProvider serverKeyProvider) {
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java]: param final

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java [hunk 5]
```diff
@@ -61,51 +56,50 @@
...
+        this.serverKeyProvider = serverKeyProvider;
+        LOG.info("using ServerKeyProvider: {}", serverKeyProvider.getClass().getName());
+    }
+    
+    @Autowired
+    public void setRegistrationStore(CaliforniumRegistrationStore registrationStore) {
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java]: param final

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java [hunk 6]
```diff
@@ -61,51 +56,50 @@
...
+        this.registrationStore = registrationStore;
+        LOG.info("using CaliforniumRegistrationStore: {}", registrationStore.getClass().getName());
     }
 
     @Autowired(required = false)
-    public void setObservationRegistryListeners(final Set<ObservationRegistryListener> listeners) {
+    public void setObservationListeners(Set<ObservationListener> listeners) {
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java]: param final

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java [hunk 7]
```diff
@@ -132,53 +126,46 @@ public void customize(final Server server) {
...
 
     @PostConstruct
-    public void startup() {
-        observationRegistry = new CaliforniumObservationRegistryImpl(observationStore,
-                clientRegistry, modelProvider, decoder);
-
-        for (ObservationRegistryListener listener : observationRegistryListeners) {
+    public void startup() throws GeneralSecurityException {
+        lwServer = createServer();
+        lwServer.start();
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java]: wouldn't it be better to register listeners first, then start?

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java [hunk 8]
```diff
@@ -132,53 +126,46 @@ public void customize(final Server server) {
...
-                .setClientRegistry(clientRegistry);
+                .setSecurityStore(securityStore);
 
         return builder.build();
     }
 
-    private void registerServlets(final ServletContextHandler root) throws GeneralSecurityException {
-
-        LeshanServer lwServer = createServer();
+    private void registerServlets(ServletContextHandler root) throws GeneralSecurityException {
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/LeshanDemoServer.java]: param final

File: adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/observation/TelemetryForwarder.java [hunk 9]
```diff
@@ -98,67 +85,69 @@ public void setEndpointMap(final Map<String, String> endpointMap) {
...
-    public void newObservation(final Observation observation) {
-
-        Client client = clientRegistry.findByRegistrationId(observation.getRegistrationId());
-        LOG.info("New observation for resource [client: {}, path: {}] has been established", client.getEndpoint(),
+    public void newObservation(final Observation observation, final Registration registration) {
+        LOG.info("New observation for resource [client: {}, path: {}] has been established", registration.getEndpoint(),
                 observation.getPath());
     }
-
+
```
Comment (sophokles73) [inline adapters/lwm2m/src/main/java/org/eclipse/hono/adapter/lwm2m/observation/TelemetryForwarder.java]: obsolete whitespace

File: core/src/test/java/org/eclipse/hono/connection/ConnectionFactoryImplTest.java [hunk 10]
```diff
@@ -72,7 +72,7 @@ public void testConnectInvokesHandlerOnfailureToConnect() throws InterruptedExce
         factory.connect(null, null, null, connectionHandler);
 
         // THEN the connection attempt fails and the given handler is invoked
-        assertTrue(latch.await(200, TimeUnit.MILLISECONDS));
+        assertTrue(latch.await(2000, TimeUnit.MILLISECONDS));
```
Comment (sophokles73) [inline core/src/test/java/org/eclipse/hono/connection/ConnectionFactoryImplTest.java]: is there a particular reason to increase this value?

Comment (DanielMaier-BSI) [inline core/src/test/java/org/eclipse/hono/connection/ConnectionFactoryImplTest.java]: My machine sems to be too slow. Test fails with 200 msec on my machine. I can also omit this change, but thought maybe others also have this problem.

**Comment by sophokles73** [comment]
Thanks, guys :-)

**Pull request merged**